packages/auto-approve/src/auto-approve.ts

```diff
@@ -56,6 +56,57 @@ export async function authenticateWithSecret(
   return new Octokit({auth: payload});
 }
 
+/**
+ * Grabs an etag for a PR, and retries adding a label if etag is different
+ *
+ * @param retryCount the amount of retries for adding a label
+ * @param owner string, of the repo of the incoming PR
+ * @param repo string, the name of the repo of the incoming PR
+ * @param prNumber number, the number of the PR
+ * @param octokit the octokit instance
+ */
+export async function retryAddLabel(
+  retryCount: number,
+  owner: string,
+  repo: string,
+  prNumber: number,
+  octokit: Octokit,
+  logger: GCFLogger = defaultLogger
+): Promise<void> {
+  const etag = (
+    await octokit.issues.listLabelsOnIssue({
+      owner,
+      repo,
+      issue_number: prNumber,
+    })
+  ).headers.etag;
+
+  try {
+    await octokit.issues.addLabels({
+      owner,
+      repo,
+      issue_number: prNumber,
+      labels: ['automerge: exact'],
+      // The comparison with the stored ETag for if-none-match uses the
+      // weak comparison algorithm, meaning two files are considered identical
+      // if the content is equivalent, not identical.
+      // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match
+      headers: {'if-none-match': `'${etag}'`},
+    });
+  } catch (err) {
+    if ((err as any).status === 412 && retryCount < 3) {
+      setTimeout(() => 2000);
```

Comment:
This should be `awaited`, however I don't know that you should even need to worry about adding a sleep (I would think the update that caused the collision updating would already be complete).